### ISSUE TITLE
False-positive message "Avoid unused local variables" one time

### ISSUE THREAD

**Issue by roukmoute** [phpmd/phpmd#797]
- PHPMD version: master
- PHP Version: 7.4.5
- Installation type: composer
- Operating System / Distribution & Version: Linux roukmoute-pc 5.4.39-rt23-MANJARO #1 SMP PREEMPT_RT Sun May 10 19:40:07 UTC 2020 x86_64 GNU/Linux

### Current Behavior
All tests passes

### Expected Behavior
First test fail, only the first time after cloned the repository.

### Steps To Reproduce:
```sh
~/workspace                                                                              
▶ git clone git@github.com:phpmd/phpmd.git
Clonage dans 'phpmd'...
remote: Enumerating objects: 35, done.
remote: Counting objects: 100% (35/35), done.
remote: Compressing objects: 100% (21/21), done.
remote: Total 19049 (delta 14), reused 35 (delta 14), pack-reused 19014
Réception d'objets: 100% (19049/19049), 4.54 Mio | 3.53 Mio/s, fait.
Résolution des deltas: 100% (11540/11540), fait.

~/workspace                                                                             ⍉
▶ cd phpmd 

~/workspace/phpmd  master ✔                                                         11d  
▶ composer install
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 42 installs, 0 updates, 0 removals
As there is no 'unzip' command installed zip files are being unpacked using the PHP zip extension.
This may cause invalid reports of corrupted archives. Besides, any UNIX permissions (e.g. executable) defined in the archives will be lost.
Installing 'unzip' may remediate them.
  - Installing symfony/polyfill-php80 (v1.17.0): Downloading (100%)         
  - Installing symfony/polyfill-ctype (v1.17.0): Downloading (100%)         
  - Installing symfony/filesystem (v5.1.0): Downloading (100%)         
  - Installing symfony/deprecation-contracts (v2.1.2): Downloading (100%)         
  - Installing symfony/config (v5.1.0): Downloading (100%)         
  - Installing psr/container (1.0.0): Loading from cache
  - Installing symfony/service-contracts (v2.1.2): Downloading (100%)         
  - Installing symfony/dependency-injection (v5.1.0): Downloading (100%)         
  - Installing pdepend/pdepend (2.7.1): Loading from cache
  - Installing psr/log (1.1.3): Loading from cache
  - Installing composer/xdebug-handler (1.4.1): Loading from cache
  - Installing myclabs/deep-copy (1.9.5): Loading from cache
  - Installing sebastian/version (2.0.1): Loading from cache
  - Installing sebastian/resource-operations (1.0.0): Loading from cache
  - Installing sebastian/recursion-context (2.0.0): Loading from cache
  - Installing sebastian/object-enumerator (2.0.1): Loading from cache
  - Installing sebastian/global-state (1.1.1): Loading from cache
  - Installing sebastian/exporter (2.0.0): Loading from cache
  - Installing sebastian/environment (2.0.0): Loading from cache
  - Installing sebastian/diff (1.4.3): Loading from cache
  - Installing sebastian/comparator (1.2.4): Loading from cache
  - Installing symfony/yaml (v4.4.9): Downloading (100%)         
  - Installing doctrine/instantiator (1.3.1): Downloading (100%)         
  - Installing webmozart/assert (1.8.0): Loading from cache
  - Installing phpdocumentor/reflection-common (2.1.0): Loading from cache
  - Installing phpdocumentor/type-resolver (1.1.0): Loading from cache
  - Installing phpdocumentor/reflection-docblock (5.1.0): Loading from cache
  - Installing phpspec/prophecy (v1.10.3): Loading from cache
  - Installing phpunit/php-text-template (1.2.1): Loading from cache
  - Installing phpunit/phpunit-mock-objects (3.4.4): Downloading (100%)         
  - Installing phpunit/php-timer (1.0.9): Loading from cache
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Loading from cache
  - Installing phpunit/php-token-stream (2.0.2): Loading from cache
  - Installing phpunit/php-file-iterator (1.4.5): Loading from cache
  - Installing phpunit/php-code-coverage (4.0.8): Loading from cache
  - Installing phpunit/phpunit (5.7.27): Downloading (100%)         
  - Installing squizlabs/php_codesniffer (2.9.2): Downloading (100%)         
  - Installing mikey179/vfsstream (v1.6.8): Loading from cache
  - Installing symfony/polyfill-mbstring (v1.17.0): Loading from cache
  - Installing gregwar/rst (v1.0.6): Downloading (100%)         
  - Installing simple-cli/simple-cli (1.4.1): Downloading (100%)         
  - Installing easy-doc/easy-doc (1.4.1): Downloading (100%)         
symfony/dependency-injection suggests installing symfony/finder (For using double-star glob patterns or when GLOB_BRACE portability is required)
symfony/dependency-injection suggests installing symfony/expression-language (For using expressions in service container configuration)
symfony/dependency-injection suggests installing symfony/proxy-manager-bridge (Generate service proxies to lazy load them)
sebastian/global-state suggests installing ext-uopz (*)
symfony/yaml suggests installing symfony/console (For validating YAML files using the lint command)
phpunit/phpunit-mock-objects suggests installing ext-soap (*)
phpunit/phpunit suggests installing phpunit/php-invoker (~1.1)
Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.
Writing lock file
Generating autoload files
12 packages you are using are looking for funding.
Use the `composer fund` command to find out more!

~/workspace/phpmd  master ✔                                                         11d  
▶ ./vendor/bin/phpunit 
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.

F..............................................................  63 / 572 ( 11%)
............................................................... 126 / 572 ( 22%)
............................................................... 189 / 572 ( 33%)
............................................................... 252 / 572 ( 44%)
............................................................... 315 / 572 ( 55%)
............................................................... 378 / 572 ( 66%)
............................................................... 441 / 572 ( 77%)
............................................................... 504 / 572 ( 88%)
............................................................... 567 / 572 ( 99%)
.....                                                           572 / 572 (100%)

Time: 4.74 seconds, Memory: 24.00MB

There was 1 failure:

1) PHPMD\Integration\CommandLineInputFileOptionTest::testReportContainsExpectedRuleViolationWarning
Failed asserting that '' contains "Avoid unused local variables such as '$foo'.".

/home/roukmoute/workspace/phpmd/src/test/php/PHPMD/Integration/CommandLineInputFileOptionTest.php:40

FAILURES!
Tests: 572, Assertions: 713, Failures: 1.
```

**Comment by AJenbo** [comment]
Solved in 3.x
